stetho/src/main/java/com/facebook/stetho/inspector/network/NetworkEventReporterImpl.java

```diff
@@ -152,6 +152,22 @@ public void responseHeadersReceived(InspectorResponse response) {
           getResourceTypeHelper().determineResourceType(contentType) :
           Page.ResourceType.OTHER;
       receivedParams.response = responseJSON;
+
+      AsyncPrettyPrinterRegistry registry= peerManager.getAsyncPrettyPrinterRegistry();
+      AsyncPrettyPrinterFactory factory = null;
+      for (int i = 0; i < response.headerCount(); i++) {
+        factory = registry.lookup(response.headerName(i));
+        if (factory != null) {
+          AsyncPrettyPrinter asyncPrettyPrinter = factory.getInstance(
+              response.headerName(i),
+              response.headerValue(i));
+          peerManager.getResponseBodyFileManager().addAsyncPrettyPrinter(
```

Comment:
The semantics don't look right here.  You aren't "adding" an async printer, you're associating it with a particular request.  Prefer `associateAsyncPrettyPrinter(...)`.